src/main/java/org/crac/Core.java

```diff
@@ -103,7 +103,7 @@ public String toString() {
         }
     }
 
-    static abstract class Compat {
```

Comment:
I'm not sure this is move in the right direction, but I'm fine with the change. In the original code Compat was a set of common code and the interface, while descendants may extend this by e.g. extra checks in constructors. While there are no extra checks now, the change looks OK, but later we may revisit this.